pkg/collector/corechecks/cluster/orchestrator/manifest_buffer.go

```diff
@@ -0,0 +1,138 @@
+	ManifestChan      chan interface{}
+	bufferedManifests []interface{}
+	stopCh            chan struct{}
+	wg                sync.WaitGroup
+}
+
+// NewManifestBuffer returns a new ManifestBuffer
+func NewManifestBuffer(chk *OrchestratorCheck) *ManifestBuffer {
+	manifestBuffer := &ManifestBuffer{
+		Cfg: &ManifestBufferConfig{
+			ClusterID:                   chk.clusterID,
+			KubeClusterName:             chk.orchestratorConfig.KubeClusterName,
+			MsgGroupRef:                 chk.groupID,
+			MaxPerMessage:               chk.orchestratorConfig.MaxPerMessage,
+			MaxWeightPerMessageBytes:    chk.orchestratorConfig.MaxWeightPerMessageBytes,
+			BufferedManifestEnabled:     chk.orchestratorConfig.BufferedManifestEnabled,
+			MaxBufferedManifests:        chk.orchestratorConfig.MaxPerMessage,
+			ManifestBufferFlushInterval: chk.orchestratorConfig.ManifestBufferFlushInterval,
+		},
+		ManifestChan: make(chan interface{}),
+		stopCh:       make(chan struct{}),
+	}
+	manifestBuffer.bufferedManifests = make([]interface{}, 0, manifestBuffer.Cfg.MaxBufferedManifests)
+
+	return manifestBuffer
+}
+
+// flushManifest flushes manifests by chunking them first then sending them to the sender
+func (cb *ManifestBuffer) flushManifest(sender aggregator.Sender) {
+	manifests := cb.bufferedManifests
+	ctx := &processors.ProcessorContext{
+		ClusterID:  cb.Cfg.ClusterID,
+		MsgGroupID: cb.Cfg.MsgGroupRef.Inc(),
+		Cfg: &config.OrchestratorConfig{
+			KubeClusterName:          cb.Cfg.KubeClusterName,
+			MaxPerMessage:            cb.Cfg.MaxPerMessage,
+			MaxWeightPerMessageBytes: cb.Cfg.MaxWeightPerMessageBytes,
+		},
+	}
+	manifestMessages := processors.ChunkManifest(ctx, manifests)
+	sender.OrchestratorManifest(manifestMessages, cb.Cfg.ClusterID)
+	cb.bufferedManifests = cb.bufferedManifests[:0]
+}
+
+// appendManifest appends manifest into the buffer
+// If buffer is full, it will flush the buffer first then append the manifest
+func (cb *ManifestBuffer) appendManifest(m interface{}, sender aggregator.Sender) {
+	if len(cb.bufferedManifests) >= cb.Cfg.MaxBufferedManifests {
+		cb.flushManifest(sender)
+	}
+
+	cb.bufferedManifests = append(cb.bufferedManifests, m)
+}
+
+// Start is to start a thread to buffer manifest and send them
+// It flushes manifests every defaultFlushManifestTime
+func (cb *ManifestBuffer) Start(sender aggregator.Sender) {
+	ticker := time.NewTicker(cb.Cfg.ManifestBufferFlushInterval)
+	cb.wg.Add(1)
+
+	go func() {
+		defer cb.wg.Done()
+	loop:
+		for {
+			select {
+			case msg, ok := <-cb.ManifestChan:
+				if !ok {
+					log.Warnf("Fail to read orchestrator manifest from channel")
+					continue
+				}
+				cb.appendManifest(msg, sender)
+			case <-ticker.C:
+				cb.flushManifest(sender)
+			case <-cb.stopCh:
+				cb.flushManifest(sender)
+				break loop
+			}
+		}
+	}()
+}
+
+// Stop is to kill the thread collecting manifest
+func (cb *ManifestBuffer) Stop() {
+	cb.stopCh <- struct{}{}
+	cb.wg.Wait()
```

Comment:
Add waitgroup here even though we only have on thread, let's make sure we flush everything when we run Stop()